test/AutoDiff/autodiff_diagnostics.swift

```diff
@@ -4,9 +4,9 @@
 // Top-level (before primal/adjoint synthesis)
 //===----------------------------------------------------------------------===//
 
-// expected-note @+1 {{opaque non-'@differentiable' function is not differentiable}}
+// expected-note @+2 {{opaque non-'@differentiable' function is not differentiable}}
```

Comment:
Why are there two of the same note?

Reply:
Could you please clarify your question?
There don't appear to be two of the same note here: there's one note and one error.
The location of the error has changed, which is unexpected (because its location didn't change in my megapatch). I'm looking into it now.

Reply:
Nvm. As you noticed, the location is not quite right, and it’s not a good diagnostic because it’s saying “expression is not differentiable” where the source location is not an expression.

Reply:
After some investigation: this is an expected temporary regression.

Currently, in `ADContext::promoteToDifferentiableFunction`, `emitAssociatedFunctionReference` is called with `parentTask == nullptr`, which leads to a poor source location for the emitted error.

```cpp
    auto assocFnAndIndices = emitAssociatedFunctionReference(
        *this, builder, /*parentTask*/ nullptr, desiredIndices, assocFnKind,
        origFnOperand, invoker, [](DifferentiationTask *newTask) {});
```

The source location will be fixed after `DifferentiationTask` is removed in favor of the `autodiff_function` design. `emitAssociatedFunctionReference` here will be called with a non-null `DifferentiationInvoker`.

---

I'll merge this patch and continue incremental AutoDiff patches. 🙂